src/app/reservation/reservation.component.ts

```diff
@@ -38,5 +43,56 @@ export class ReservationComponent {
     });
   }
 
+  acceptReservation() {
+    if (this.reservation && this.book && this.user) {
+
+      // get bookcopy data for the specific book
+      this.bookCopyService.getBookCopy(this.book.id).subscribe(bookCopyResponse => {
+        const bookCopyData: ReadBookCopyDto[] = bookCopyResponse.data;
+
+        // check if bookCopy is available and find first bookCopy
+        const availableBookCopy = bookCopyData.find(copy => copy.available);
 
-}
+        if (availableBookCopy) {
+          // change reservationRequest status
+          if (this.reservation?.reservationRequest == "PENDING") {
+            this.reservation.reservationRequest = "ACCEPTED"
+            this.reservationService.updateReservation(this.reservation).subscribe(response => {
+              console.log('Reservation updated:', response);
+            });
+          }
+
+          // create loan
+          const loanData = {
+            conditionStart: availableBookCopy.state,
+            startDate: new Date(),
+            bookCopyId: availableBookCopy.id,
+            isActive: true,
+            userId: this.user?.id,
+            bookId: this.book?.id
+          };
+
+          this.loanService.createLoan(loanData).subscribe(response => {
+            console.log('Loan created:', response);
+            alert("Lening aangemaakt")
```

Comment:
Nadat je een reservation accept word de loan lijst niet meteen geupdate en moet je eerst refreshen.
Bij book catalogue is het geimplementeerd dat het automatisch de nieuw toegevoegde items ophaalt.
Volgensmij kan dat hier in de subscribe van lening aanmaken.

Reply:
Dit lukt me nu niet.